src/workerd/util/sqlite.c++

```diff
@@ -182,19 +217,182 @@ kj::Own<sqlite3_stmt> SqliteDatabase::prepareSql(sqlite3* db, kj::StringPtr sqlC
+    case SQLITE_DELETE             :   /* Table Name      NULL            */
+    case SQLITE_DROP_TABLE         :   /* Table Name      NULL            */
+    case SQLITE_INSERT             :   /* Table Name      NULL            */
+    case SQLITE_ANALYZE            :   /* Table Name      NULL            */
+    case SQLITE_CREATE_VIEW        :   /* View Name       NULL            */
+    case SQLITE_DROP_VIEW          :   /* View Name       NULL            */
+    case SQLITE_REINDEX            :   /* Index Name      NULL            */
+      KJ_ASSERT(param2 == nullptr);
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1));
+
+    case SQLITE_ALTER_TABLE        :   /* Database Name   Table Name      */
+      // Why is the database name passed redundantly here? Weird.
+      KJ_ASSERT(KJ_ASSERT_NONNULL(param1) == "main"_kj);
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
+
+    case SQLITE_READ               :   /* Table Name      Column Name     */
+    case SQLITE_UPDATE             :   /* Table Name      Column Name     */
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1));
+
+    case SQLITE_CREATE_INDEX       :   /* Index Name      Table Name      */
+    case SQLITE_DROP_INDEX         :   /* Index Name      Table Name      */
+    case SQLITE_CREATE_TRIGGER     :   /* Trigger Name    Table Name      */
+    case SQLITE_DROP_TRIGGER       :   /* Trigger Name    Table Name      */
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param1)) &&
+             regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
+
+    case SQLITE_TRANSACTION        :   /* Operation       NULL            */
+      {
+        // Verify param1 is one of the values we expect.
+        kj::StringPtr op = KJ_ASSERT_NONNULL(param1);
+        KJ_ASSERT(op == "BEGIN" || op == "ROLLBACK" || op == "COMMIT", op);
+      }
+      KJ_ASSERT(param2 == nullptr);
+      return true;
+
+    case SQLITE_SAVEPOINT          :   /* Operation       Savepoint Name  */
+      {
+        // Verify param1 is one of the values we expect.
+        kj::StringPtr op = KJ_ASSERT_NONNULL(param1);
+        KJ_ASSERT(op == "BEGIN" || op == "ROLLBACK" || op == "COMMIT", op);
+      }
+      return regulator.isAllowedName(KJ_ASSERT_NONNULL(param2));
+
+    case SQLITE_PRAGMA             :   /* Pragma Name     1st arg or NULL */
+      // TODO(sqlite): Decide which pragmas are OK.
+      return false;
+
+    case SQLITE_FUNCTION           :   /* NULL            Function Name   */
+      // TODO(sqlite): Decide which function are OK.
+      return false;
+
+    // ---------------------------------------------------------------
+    // Stuff that is never allowed
+
+    case SQLITE_CREATE_VTABLE      :   /* Table Name      Module Name     */
+    case SQLITE_DROP_VTABLE        :   /* Table Name      Module Name     */
+      // Virtual tables are tables backed by some native-code callbacks. We don't support these.
+      return false;
```

Comment:
We might be able to support this safely in future, where a user could define a virtual table in JS and SQLite could call into it, potentially even connecting it to some external data source. Opens the whole async can of worms though, so happy to disable for now.